app/models/discussion.rb

```diff
@@ -8,12 +8,16 @@ class Discussion < ActiveRecord::Base
 
   default_scope -> { order('updated_at DESC') }
 
+  after_create :find_or_create_subscriptions
   after_touch :notify_new_comment
 
   private
 
+  def find_or_create_subscriptions
+    Subscriber.for_discussion(self).subscribe(User.all)
+  end
+
   def notify_new_comment
-    subscriptions = Subscriber.for_discussion(self).subscribe(User.all)
-    subscriptions.map(&:new_comment)
+    find_or_create_subscriptions.map(&:new_comment)
   end
 end
```

Comment:
@naps62 Doesn't it change state on any comment? It's on the after touch, and it updates all the subscriptions (and creates new ones, when they don't exist). Where is the logic that makes it only change when it's your comment?